cli/tests/perconaMySqlServer.spec.ts

```diff
@@ -0,0 +1,449 @@
+      await output.outContains('MySQL Service added.');
+    }
+  });
+
+  /**
+   * @link https://github.com/percona/pmm-qa/blob/main/pmm-tests/pmm-2-0-bats-tests/ps-specific-tests.bats#L347
+   */
+  test('run pmm-admin remove mysql added via metrics mode pull', async ({ }) => {
+    let hosts = (await cli.exec(`sudo pmm-admin list | grep "MySQL" | grep "mysql_"`))
+      .stdout.trim().split('\n').filter((item) => item.trim().length > 0);
+    let n = 1;
+    for (const host of hosts) {
+      let output = await cli.exec(`sudo pmm-admin remove mysql mysql_${n++}`);
+      await output.assertSuccess();
+      await output.outContains('Service removed.');
+    }
+  });
+
+  /**
+   * @link https://github.com/percona/pmm-qa/blob/main/pmm-tests/pmm-2-0-bats-tests/ps-specific-tests.bats#L359
+   */
+  test("PMM-T160 User can't use both socket and address while using pmm-admin add mysql", async ({ }) => {
+    let hosts = (await cli.exec(`sudo pmm-admin list | grep "MySQL" | awk -F" " '{print $3}'`))
+      .stdout.trim().split('\n').filter((item) => item.trim().length > 0);
+    let n = 1;
+    for (const host of hosts) {
+      const mysql_ip = host.split(':')[0];
+      const mysql_port = host.split(':')[1];
+      let output = await cli.exec(`sudo  pmm-admin add mysql --query-source=perfschema --username=${MYSQL_USER} --password=${MYSQL_PASSWORD} --host=${mysql_ip} --socket=/tmp/mysql_sandbox${mysql_port}.sock --service-name=mysql_${n++}`);
+      await output.exitCodeEquals(1);
+      await output.outContains('Socket and address cannot be specified together.');
+    }
+  });
+
+  /**
+   * @link https://github.com/percona/pmm-qa/blob/main/pmm-tests/pmm-2-0-bats-tests/ps-specific-tests.bats#L375
+   */
+  test("PMM-T159 User can't use both socket and port while using pmm-admin add mysql", async ({ }) => {
+    let hosts = (await cli.exec(`sudo pmm-admin list | grep "MySQL" | awk -F" " '{print $3}'`))
+      .stdout.trim().split('\n').filter((item) => item.trim().length > 0);
+    let n = 1;
+    for (const host of hosts) {
+      const mysql_ip = host.split(':')[0];
+      const mysql_port = host.split(':')[1];
+      let output = await cli.exec(`sudo  pmm-admin add mysql --query-source=perfschema --username=${MYSQL_USER} --password=${MYSQL_PASSWORD} --port=${mysql_port} --socket=/tmp/mysql_sandbox${mysql_port}.sock --service-name=mysql_${n++}`);
+      await output.exitCodeEquals(1);
+      await output.outContains('Socket and port cannot be specified together.');
+    }
+  });
+
+  /**
+   * @link https://github.com/percona/pmm-qa/blob/main/pmm-tests/pmm-2-0-bats-tests/ps-specific-tests.bats#L392
+   */
+  test("PMM-T789 - Verify help for pmm-admin add mysql has TLS-related flags", async ({ }) => {
+    const output = await cli.exec(`sudo pmm-admin add mysql --help`);
+    await output.assertSuccess();
+    await output.trimWhitespace();
+    await output.outContains('tls Use TLS to connect to the database');
```

Comment:
if test will fail on the first assert and the following will not be executed, please use containsMany() for multiline assert, see: https://github.com/percona/pmm-ui-tests/blob/main/cli/tests/help.spec.ts#L39

Also, do you wish to move given test to the help spec and apply your tab normalization method improvement to the whole help spec? It seems that method name is misleading - I read it as "remove whitepaces at all" instead of "reduce whitespace fill between console out put columns"

Reply:
Changed method and created new for verifying normalized which removes unnecessary characters.

Reply:
thank you